packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -106,44 +195,39 @@ export class VaDate {
       error,
       handleDateBlur,
       handleDateChange,
+      handleDateKey,
       value,
     } = this;
 
-    // Convert string to number to remove leading 0 on values less than 10
-    const [year, month, day] = (value || '')
-      .split('-')
-      .map(val => parseInt(val, 10));
-
-    const daysForSelectedMonth = month > 0 ? days[month] : [];
-
-    // Check validity of date if invalid provide message and error state styling
-    const dateInvalid =
-      required && (!isFullDate(value) || day > daysForSelectedMonth.length)
-        ? 'Please provide a valid date'
-        : null;
+    const [year, month, day] = (value || '').split('-').map(val => val);
+    const monthNum = parseInt(month);
+    const dayNum = parseInt(day);
```

Comment:
Why is the `.map()`/`parseInt()` behavior changing here?

Reply:
It should not you are correct and I have updated accordingly to fix this issue.